lib/validators/helpers.js

```diff
@@ -30,6 +221,92 @@ export function getPreviewText(text, index, length) {
   return previewText;
 }
 
+/**
+ * detrmine whether a conditional-required field should be required
+ * note: exported for testing
+ * @param  {object} field         { type, name, path, schema, value }
+ * @param  {object} componentData
+ * @return {boolean}
+ */
+export function shouldBeRequired(field, componentData) {
+  const compareField = _.get(field, 'validate.required.field'),
+    compareOperator = _.get(field, 'validate.required.operator'),
+    compareValue = _.get(field, 'validate.required.value'),
+    path = _.get(field, 'path'),
+    pathArray = path.split('.');
+
+  let comparePath;
+
+  if (pathArray.length > 1) {
+    // iterate up through the path, checking to see if the parent item (for complex lists)
+    // has the field in question
+    _.findLast(pathArray, (pathPart, pathIndex) => {
+      if (pathPart.match(/\d+/)) {
+        const subPath = pathArray.slice(0, pathIndex + 1).join('.');
+
+        if (_.has(componentData, `${subPath}.${compareField}`)) {
+          comparePath = `${subPath}.${compareField}`;
+          return true;
+        } else {
+          return false;
+        };
+      } else {
+        return false;
+      }
+    });
+  } else {
+    comparePath = compareField;
+  }
+
+  return compare({ data: _.get(componentData, comparePath), operator: compareOperator, value: compareValue });
+}
+
+/**
+ * determine if a field passes built-in validation
+ * @param  {object}  field         { type, name, path, schema, value }
+ * @param  {object}  componentData
+ * @param {string} [type] if you only want to validate against a specific thing
+ * @return {Boolean}
+ */
+export function isValid(field, componentData, type) { // eslint-disable-line
+  const validate = _.get(field, 'validate'),
+    fieldValue = _.get(field, 'value'),
+    isFieldEmpty = isEmpty(fieldValue),
+    strValue = _.isString(fieldValue) && getPlaintextValue(fieldValue);
+
+  let length;
+
+  if (!_.isObject(validate)) {
+    return true; // if no built-in validation rules, return true
+  }
+
+  // if we're dealing with strings or arrays, we'll want to check min/max against the length
+  // (if it's numbers, dates, or other things, we want to check min/max against the value directly)
+  if (strValue || _.isArray(fieldValue)) {
+    length = (strValue || fieldValue).length;
+  }
+
+  if (type) {
+    switch (type) {
+      case 'required': return !(validate.required === true && isFieldEmpty);
+      case 'conditional-required': return !(_.isObject(validate.required) && shouldBeRequired(field, componentData) && isFieldEmpty);
+      case 'pattern': return !(!isFieldEmpty && validate.pattern && !strValue.match(new RegExp(validate.pattern, 'ig')));
+      case 'min': return !(validate.min && (length || fieldValue) < validate.min);
+      case 'max': return !(validate.max && (length || fieldValue) > validate.max);
+      default: throw new Error(`Unsupported validation type: ${type}`);
```

Comment:
I don't want it to log and continue, because this is a dev error that should (hopefully) never be hit unless someone's

1. in the process of writing a validator, that
2. erroneously assumes there's some kind of built-in validation that doesn't exist